tests/atoms/gaia_apps.js

```diff
@@ -181,20 +181,26 @@ var GaiaApps = {
     var apps = window.wrappedJSObject.applications ||
                 window.wrappedJSObject.Applications;
     var app = apps.getByManifestURL(manifestURL);
-    var appName, launchPath;
+    var launchPath;
 
-    if (entryPoint) {
-      if (app.manifest.entry_points[entryPoint]) {
-        appName = app.manifest.entry_points[entryPoint].name;
-        launchPath = app.manifest.entry_points[entryPoint].launch_path;
-      } else {
-        app = null;
-      }
+    if (app != null) {
+      GaiaApps._formatApp(app, manifestURL, entryPoint, callback, launchPath);
```

Comment:
I guess Johan told you to put this in a separate function?
I'd rather keep it as part of this function, because it's only used once, but if he prefers it that way, I'm fine with it.